### ISSUE TITLE
Windows compatibility

### ISSUE THREAD

**Issue by GoogleCodeExporter** [simpcomp-team/simpcomp#1]
```
Some tools used by simpcomp are not available on Windows (e.g. uname) - update 
source accordingly. Also provide a .zip release for Windows users.
```

Original issue reported on code.google.com by `effenber...@gmail.com` on 22 Jan 2013 at 7:37

**Comment by fingolfin** [comment]
Does simpcomp work on Windows / Cygwin? Maybe @ChrisJefferson did try it...

I do see that `uname` is used in the GAP code to obtain the hostname... but in the end this is really only used for the function `SCMailSend`, which probably won't work on Windows for other reasons, but also doesn't seem essentially for the workings of simpcomp?

@simpcomp-team perhaps you could simply remove `SCIntFunc.GetSystemType` and instead use `GAPInfo.Architecture`, which arguably is more useful (contains info what GAP knows/assumes about the system), while including less private irrelevant info compared to `uname -a`.

(BTW, the use of "master" in that greeting text of that email doesn't seem appropriate for this day and age, I'd personally just drop it).

**Comment by ChrisJefferson** [comment]
`uname` is fine. `mail` is never going to work on Windows (but I imagine it also doesn't work any more on many linux systems?)

The package does complain `homology` is not installed when I run the tests, but that package generally doesn't seem to be distributed with GAP?

Also, some tests (at least on my machine) fail, but that is because they are printing groups out in a different style (for example I get:

```
########> Diff in /opt/gap-master/pkg/simpcomp--2.1.12/tst/simpcomp.tst:181
# Input is:
SCAutomorphismGroup(c);
# Expected output:
Group([ (1,5,3)(2,6,4)(7,8), (1,7,3,5)(2,8,4,6) ])
# But found:
Group([ (7,8), (5,7)(6,8), (3,5)(4,6), (1,3)(2,4) ])
```

I would re-write the test to `SCAutomorphismGroup(c) = Group([ (1,5,3)(2,6,4)(7,8), (1,7,3,5)(2,8,4,6) ])`, and then check the answer is `true`, but I imagine that isn't cygwin-specific.

**Comment by fingolfin** [comment]
Yes, the complain about `homology` is sadly normal. It exists at https://linalg.org/gap.html but AFAIK it was never deposited or submitted for distribution (but then again it is 20 years old, which is long before "my time" with GAP started, so perhaps it was submitted... and somehow this never worked out?!? hard to tell for me...). Anyway, the download link there is broken. I guess one could try to contact the authors to figure out the state of this

The different groups may be caused by `grape` not being loaded / loadable, at least then I got similar diffs.

**Comment by fingolfin** [comment]
Ah it seems a newer version is available here: https://membres-ljk.imag.fr/Jean-Guillaume.Dumas/Homology/

**Comment by fingolfin** [comment]
Aha, it requires a Pascal compiler :-). I don't think I'll go down that road just now... :-)

**Comment by simpcomp-team** [comment]
A long time ago I heard from the authors of `homology` that they did try and submit it but never implemented comments from the code review (the package contains a probabilistic algorithm and there were some comments about this - but I don't know the details). In any case, it is very fast, so we kept it as the preferred method of computing homology but have our own algorithms to fall back on.

**Comment by simpcomp-team** [comment]
I'll adjust the mailing program (and the greeting) for the next release. I am thinking about removing the whole mailing interface because it hardly every works in practice (either too many emails or not enough).

**Comment by simpcomp-team** [comment]
Leaving this open until I make a final decision on email functionality.
